src/rpc.rs

```diff
@@ -307,3 +337,16 @@ pub fn cell_count_for_confidence(confidence: f64) -> u32 {
 	}
 	cell_count
 }
+
+pub async fn check_version(url: &str) -> Result<bool> {
+	let version = get_chain(url).await?;
+	let runtime_version = get_runtime_version(url).await?;
+	if version != "1.1.0"
+		|| runtime_version.spec_version != 5
+		|| runtime_version.spec_name != "data-avail"
+	{
+		return Err(anyhow!("Invalid version"));
+	} else {
```

Comment:
Instead of having both network errors and invalid version under the Err, we probably should have `Ok(true)` for valid, `Ok(false)` for invalid, and `Err` for other cases, since client behavior would be different in all three cases?

Reply:
Also, maybe it would be cool to have 
```
fn version_matches(version: String, runtime_version: RuntimeVersion) -> bool {
    version != "1.1.0" || runtime_version.spec_version != 5 || runtime_version.spec_name != "data-avail"
}```

And leave errors and side effects to upper layer:

```
let version = get_chain(url).await?;
let runtime_version = get_runtime_version(url).await?;
if version_matches(version, runtime_version) {
    ...do the thing
}```

Or even with proper typing:
```
let version = get_chain(url).await?;
let runtime_version = get_runtime_version(url).await?;
if version.not_below("1.1.0") && runtime_version.not_below(5, "data_avail") {
  // do the thing
}
```